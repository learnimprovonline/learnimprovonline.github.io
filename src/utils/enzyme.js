import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-15'
import toJson from 'enzyme-to-json'

Enzyme.configure({ adapter: new Adapter() })
