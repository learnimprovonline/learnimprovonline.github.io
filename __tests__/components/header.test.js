import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Link from 'gatsby-link'
import Header from '../../src/components/header'

const HeaderDom = shallow(<Header>{'Site Name'}</Header>)

describe('Header', () => {

  describe('Site Title', () => {
    const siteTitle = HeaderDom.find('.header-logo')
    const titleLink = siteTitle.find(Link)

    test('link goes to the main page', () => {
      const rootDestination = '/'

      const linkDestination = titleLink.prop('to')
      expect(linkDestination).toEqual(rootDestination)
    })

    test('display text is passed in as children', () => {
      const linkText = titleLink.children().text()

      expect(linkText).toEqual('Site Name')
    })
  })

  describe('Navigation', () => {
    const navigationItems = HeaderDom.find('.header-nav-list-item')

    test('should have three navigation items', () => {
      expect(navigationItems).toHaveLength(3)
    })

    describe('Activities', () => {
      const activitiesLink = navigationItems.at(0).find(Link)

      test('link goes to Activities page', () => {
        const activitesPageDestination = '/activities'
        const linkDestination = activitiesLink.prop('to')

        expect(linkDestination).toEqual(activitesPageDestination)
      })
      test('display text is "Activities"', () => {
        const linkText = activitiesLink.children().text()

        expect(linkText).toEqual('Activities')
      })
    })

    describe('FAQ', () => {
      const faqLink = navigationItems.at(1).find(Link)

      test('link goes to FAQ page', () => {
        const faqPageDestination = '/faq'
        const linkDestination = faqLink.prop('to')

        expect(linkDestination).toEqual(faqPageDestination)
      })
      test('display text is "FAQ"', () => {
        const linkText = faqLink.children().text()

        expect(linkText).toEqual('FAQ')
      })
    })

    describe('Contact', () => {
      const contactLink = navigationItems.at(2).find(Link)

      test('link goes to Contact page', () => {
        const contactPageDestination = '/contact'
        const linkDestination = contactLink.prop('to')

        expect(linkDestination).toEqual(contactPageDestination)
      })
      test('display text is "Contact"', () => {
        const linkText = contactLink.children().text()

        expect(linkText).toEqual('Contact')
      })
    })
  })

  test('renders itself correctly', () => {
    const tree = toJson(HeaderDom)

    expect(tree).toMatchSnapshot()
  })
})
