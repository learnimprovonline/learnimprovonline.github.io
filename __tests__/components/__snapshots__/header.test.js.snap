// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders itself correctly 1`] = `
<nav
  class="navbar navbar-expand-lg navbar-dark bg-dark"
>
  <div
    class="container"
  >
    <GatsbyLink
      className="navbar-brand"
      to="/"
    >
      Site Name
    </GatsbyLink>
    <button
      aria-controls="navbarNavAltMarkup"
      aria-expanded="false"
      aria-label="Toggle navigation"
      class="navbar-toggler"
      data-target="#navbarNavAltMarkup"
      data-toggle="collapse"
      type="button"
    >
      <span
        class="navbar-toggler-icon"
      />
    </button>
    <div
      class="collapse navbar-collapse"
      id="navbarNavAltMarkup"
    >
      <div
        class="navbar-nav"
      >
        <GatsbyLink
          className="nav-item nav-link"
          to="/activities"
        >
          Activities
        </GatsbyLink>
        <GatsbyLink
          className="nav-item nav-link"
          to="/faq"
        >
          FAQ
        </GatsbyLink>
        <GatsbyLink
          className="nav-item nav-link"
          to="/contact"
        >
          Contact
        </GatsbyLink>
      </div>
    </div>
  </div>
</nav>
`;
